### PR TITLE
Add Support for GitHub Copilot Seat Management in Organizations

### PR DESCRIPTION
**Pull request by pashafateev**
### Summary
This pull request introduces support for managing GitHub Copilot seats in organizations. It adds new classes and methods to handle Copilot-related API calls, such as retrieving all seats, adding new seats, and removing seats for specific users.

### Motivation
The addition of GitHub Copilot seat management functionality extends PyGithub to include support for the Copilot Business API. This enhancement enables users to automate and manage Copilot seats within their organizations programmatically, improving workflow efficiency.

### Changes
- **New Classes**:
  - `Copilot`: Represents the Copilot entity and provides methods to interact with Copilot-related endpoints.
  - `CopilotSeat`: Represents individual Copilot seats and their attributes, such as assignee, last activity, and plan type.

- **New Methods**:
  - `Organization.get_copilot`: Retrieves a `Copilot` object for the organization.
  - `Copilot.get_seats`: Fetches all Copilot seats for an organization.
  - `Copilot.add_seats`: Adds seats for selected usernames.
  - `Copilot.remove_seats`: Removes seats for selected usernames.

### CODE REVIEW

File: github/Copilot.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
+from __future__ import annotations
```
Comment (EnricoMi) [inline github/Copilot.py:1]: Please add the licence header.

File: github/Copilot.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+            self._org_name = self._makeStringAttribute(attributes["org_name"])
+
+    def __repr__(self) -> str:
+        return self.get__repr__({"org_name": self._org_name.value if self._org_name is not NotSet else NotSet})
+
+    @property
+    def org_name(self) -> str:
+        return self._org_name.value
+
+    def get_all_seats(self) -> PaginatedList[CopilotSeat]:
```
Comment (EnricoMi) [inline github/Copilot.py:32]:
```suggestion
    def get_seats(self) -> PaginatedList[CopilotSeat]:
```

File: github/Copilot.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        url = f"/orgs/{self._org_name.value}/copilot/billing/seats"
+        return PaginatedList(
+            github.CopilotSeat.CopilotSeat,
+            self._requester,
+            url,
+            None,
+            list_item="seats",
+        )
+
+    def add_seat(self, selected_usernames: list[str]) -> int:
```
Comment (EnricoMi) [inline github/Copilot.py:45]:
```suggestion
    def add_seats(self, selected_usernames: list[str]) -> int:
```

File: github/Copilot.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+        """
+        url = f"/orgs/{self._org_name.value}/copilot/billing/selected_users"
+        _, data = self._requester.requestJsonAndCheck(
+            "POST",
+            url,
+            input={"selected_usernames": selected_usernames},
+        )
+        return data["seats_created"]
+
+    def remove_seat(self, selected_usernames: list[str]) -> int:
```
Comment (EnricoMi) [inline github/Copilot.py:60]:
```suggestion
    def remove_seats(self, selected_usernames: list[str]) -> int:
```

**Review by EnricoMi — COMMENTED** [review]
LGTM! Minor comments.

File: github/Copilot.py (hunk 1 above)
Comment (pashafateev) [reply github/Copilot.py:23]: Done

**Review by EnricoMi — COMMENTED** [review]
More changes needed, replay data files also need to be renamed (because test names have been changed).

**Comment by EnricoMi** [comment]
Please check the test and lint failures.

**Comment by pashafateev** [comment]
> Please check the test and lint failures.

ok, I think the last commit should address the failing tests.

**Review by EnricoMi — APPROVED** [review]
LGTM!

**Comment by pashafateev** [comment]
This is my first contribution, so just want to make sure. Is my work here done? You will merge from here?

**Comment by EnricoMi** [comment]
Yes, all good, thanks for your contribution!

**Pull request merged**